### PR TITLE
Test hwtimer wo sleep

### LINKED ISSUE #1204: tests: Examine the need for thread_sleep() in test_hwtimer
In #1083 a `thread_sleep()` at the end of `test_hwtimer` was introduced to assure that a broken `thread_exit()` for any platform would not interfere with intended testing functionality. However, this is suboptimal and prevents the test from terminating which is desirable for automatic testing.

### PR DESCRIPTION
**Pull request by OlegHahm**
Fixes #1204,

### CODE REVIEW

**Comment by OlegHahm** [comment]
Oops, made a mistake. Will fix.

**Comment by OlegHahm** [comment]
fixed

File: tests/hwtimer/main.c [hunk 1]
```diff
@@ -25,10 +25,12 @@
 #include "hwtimer.h"
 #include "thread.h"
```
Comment (PeterKietzmann) [inline tests/hwtimer/main.c]: Now that you are at it: With removing `thread_sleep();` this include gets superfluous right?

Comment (OlegHahm) [inline tests/hwtimer/main.c]: Yes, you're right. Addressed.

**Comment by PeterKietzmann** [comment]
@kaspar030 do you have an opinion here? I'd give my ACK but maybe I didn't get the "Request for Comments" point here. @OlegHahm IMO you can already squash and we run Travis. The PR is small and in case @kaspar030 ACKs he can just merge it.

**Comment by OlegHahm** [comment]
squashed

**Pull request merged**